controller/src/main/java/org/jboss/as/controller/ModelControllerImpl.java

```diff
@@ -518,7 +518,7 @@ boolean boot(final List<ModelNode> bootList, final OperationMessageHandler handl
                     headers, handler, null, managementModel.get(), control, processState, auditLogger,
                             bootingFlag.get(), true, hostServerGroupTracker, null, notificationSupport, true,
                             extraValidationStepHandler, partialModel, securityIdentitySupplier);
-            if (configExtension != null && configExtension.shouldProcessOperations(runningModeControl.getRunningMode())) {
+            if (configExtension != null && configExtension.shouldProcessOperations(runningModeControl.getRunningMode()) && (!runningModeControl.isReloaded() || runningModeControl.isApplyConfigurationExtension())) {
```

Comment:
I think we should simply do this, it is difficult to track down and it took me some time to understand its meaning.

My feeling is this logic should be encapsulated inside  `configExtension.shouldProcessOperations`. I think the additional `&& (!runningModeControl.isReloaded() || runningModeControl.isApplyConfigurationExtension())` complements it by adding a new case to shouldProcessOperations.

Maybe shouldProcessOperations to take into account the whole `runningModeControl` instead of only the `runningModeControl.getRunningMode()`